README.md

```diff
@@ -12,125 +13,145 @@ npm install @girder/components
+});
+
+// This is passed to our Vue instance; it will be available in all components
+const GirderProvider = {
+  girderRest,
+};
+export default GirderProvider;
+```
+
+Reference the configuration from your application initialization (typically `src/main.js`):
 
 ```javascript
-import * as girder from '@girder/components';
-// apiRoot should point to your girder instance
-const restClient = new girder.RestClient({ apiRoot: 'http://localhost:8080/api/v1' });
+import GirderProvider from '@/plugins/girder';
+
+// ...
 
-// Provide "girderRest" to the root
 new Vue({
-  provide: { girderRest },
-  ...
+  provide: GirderProvider,
+  // ...
 }).$mount('#app');
 ```
 
-#### Components
-
-If you're building your own downstream application, you can include individual components from the library. Because these are vuetify components, your consumer application is responsible for creating the `v-app` container.  [Read more](https://vuetifyjs.com/en/layout/pre-defined#all-about-app).  See `demo/main.js` for a more comprehensive example.
+### Using Components
 
-Either import the full UMD module:
+Components should be imported from `@girder/components/src/components`, as this path will be stable across releases.
 
-```javascript
-import Girder, * as girder from '@girder/components';
-```
+Since Girder web components uses Vuetify, your application must provide
+[`v-app` and `v-content` containers](https://vuetifyjs.com/en/framework/default-markup#all-about-app)
+at some ancestor level.
 
-Or import specific components from the `src` directory:
+For example, to create a login / registration widget in `src/App.vue`:
 
 ```html
-<!-- Downstream.vue -->
 <template>
-    <girder-upload ... />
+  <v-app>
+    <v-content>
+      <GirderAuthentication register />
+    </v-content>
+  </v-app>
 </template>
 
 <script>
-import { Upload as GirderUpload } from '@girder/components/src/components';
+import { Authentication as GirderAuthentication } from '@girder/components/src/components';
 
 export default {
-    components: { GirderUpload },
-    ...
+  components: {
+    GirderAuthentication,
+  },
+
+  // Injecting is not necessary to use the component,
+  // but is required to access the results of API calls
+  inject: ['girderRest'],
+  created() {
+    this.girderRest.$on('login', this.onLogin);
+  },
+  methods: {
+    onLogin(user) {
+      console.log('Login succeeded for', user);
+    },
+  },
 };
 </script>
 ```
 
-> **Note:** When importing components from the source tree, you should import
-> them from `index.js` rather than importing the `.vue` files yourself, as the
-> latter is prone to break if files get moved in future releases. For example:
-> ```javascript
->  import { Upload as GirderUpload } from '@girder/components/src/components';  // Good
->  import GirderUpload from '@girder/components/src/components/Upload.vue'; // Unsafe -- may move in future
-> ```
-> Files and symbols that do not appear in an index.js should be considered private and it
```

Comment:
Are we intentionally getting rid of this normative bit about not using symbols unless they're exported all the way to the top level of the UMD module?

Reply:
It's now simplified above:
> Components should be imported from `@girder/components/src/components`, as this path will be stable across releases.

Reply:
I think that may be a little vague.  `@girder/components/src/components` could be interpreted as a base path, and when I'm looking through READMEs, I usually skim over sentences and read code examples.

I'd prefer to keep this section as-is.

Reply:
I'll pull the code examples back into this section. I agree, we don't want to make the text longer.